R/thumbnail.R

```diff
@@ -0,0 +1,233 @@
+#' Get the Content Thumbnail
+#'
+#' `get_thumbnail` saves the content thumbnail to the given path (default: temp file).
+#' `delete_thumbnail` removes the thumbnail (optionally saving to the given path)
+#' `has_thumbnail` returns whether the content has an thumbnail
+#'
+#' @param content A content object
+#' @param path optional. The path to the image on disk
+#'
+#' @rdname get_thumbnail
+#' @family content functions
+#' @export
+get_thumbnail <- function(content, path = NULL) {
```

Comment:
I don't understand the use case for `path = NULL`. What if `path` were a required argument, and if someone wanted to write to a temp file, they can explicitly say `path = tempfile()`. That's better anyway so that the caller can manage the lifetime of that temp file (as we should in tests).